src/components/forms/OpaqueIngressSecretForm.js

```diff
@@ -84,9 +95,17 @@ class OpaqueIngressSecretForm extends Component {
             onChange={this.handleInputChange}
             placeholder="my-opaque-ingress-name"
             invalidCallback={() =>
-              this.props.secretCallback(this.state, this.props)
+              this.props.secretCallback("opaque_secrets")(
+                this.state,
+                this.props,
+                "name",
+              )
             }
-            invalidText={this.props.secretCallbackText(this.state, this.props)}
+            invalidText={this.props.secretCallbackText("opaque_secrets")(
```

Comment:
we should not be hardcoding `"opaque_secrets"` here, this should be a function that accepts two parameters `stateData` and `componentProps`

Reply:
we can have a third prop for name that can be hard coded